source/aggregation-pipeline-builder.txt

```diff
@@ -89,69 +89,226 @@ Limitations
+
+            You can adjust the width of the :guilabel:`Stage Input`, stage 
+            editor, and the :guilabel:`Stage Output` by dragging their border to 
+            the desired size.
+
+            .. note:: 
+
+               .. include:: /includes/fact-agg-pipeline-preview-toggle.rst
+
+         .. step:: Add additional pipeline stages.
+            
+            Click the :guilabel:`Add Stage` dropdown to add additional 
+            aggregation stages before or after your last aggregation stage. 
+            Repeat steps 4 and 5 for each additional stage.
+
+            You can add stages with the following keyboard shortcuts: 
+
+            - To add a stage after the current stage: 
+
+              - Windows / Linux: :guilabel:`Ctrl + Shift + A`
+
+              - Mac: :guilabel:`⌘ + Shift + A`
+            
+            - To add a stage before the current stage: 
+              
+              - Windows / Linux: :guilabel:`Ctrl + Shift + B`
+
+              - Mac: :guilabel:`⌘ + Shift + B` 
+
+         .. step:: Navigate between stages. 
+            
+            To navigate between different stages, select the stage you 
+            want to edit from the :guilabel:`Stage` dropdown in the upper-right 
+            corner of the Focus Mode modal. 
+
+            You can navigate between stages with the following keyboard 
+            shortcuts: 
+
+            - To go to the stage before the current stage: 
+
+              - Windows / Linux: :guilabel:`Ctrl + Shift + 9`
+
+              - Mac: :guilabel:`⌘ + Shift + 9`
+            
+            - To add a stage before the current stage: 
+              
+              - Windows / Linux: :guilabel:`Ctrl + Shift + 0`
+
+              - Mac: :guilabel:`⌘ + Shift + 0` 
+
+         .. step:: Run the pipeline.
+            
+            Click :guilabel:`x` to exit Focus Mode and select 
+            :guilabel:`Run` at the top right of the pipeline builder. 
+            |compass-short| returns your results in the document view. 
+
+   .. tab:: Text View Mode 
+      :tabid: text 
+
+      .. procedure::
+         :style: normal
+         
+         .. step:: Select the :guilabel:`Text` view.
+            
+            In the aggregation pipeline pane, click the 
+            :guilabel:`</> Text` toggle switch to enable text mode for pipeline 
+            editing.
+            
+            .. figure:: /images/compass/agg-builder-textview-toggle-on.png
+               :alt: Toggle textmode on
+         
+         .. step:: Enter your aggregation pipeline.
+            
+            Enter valid aggregation syntax in the text editor. The 
+            following example pipeline limits the query results to ``4`` 
+            documents.
+            
+            .. code-block:: javascript
+               
+               [ { "$limit" : 4 } ]
+               
+            .. note:: 
+               
+               **(Optional)** If preview mode is enabled, you can click 
```

Comment:
[suggestion]

Because this is no longer a step, I'm not sure we need to call out that this is optional. The fact that it's in a `note` already kind of implies that it's an aside.